### PR TITLE
Add release checklist

### PR DESCRIPTION
**Pull request by renefritze**

### CODE REVIEW

**Comment by renefritze** [comment]
This closes #534

**Comment by pmli** [comment]
To get a checklist like

- [ ] one
- [ ] two

it seems necessary to use `- [ ]`:

```
- [ ] one
- [ ] two
```

BTW, why not use a numbered list?

Maybe it would be good to reference it somewhere (e.g., `README.md` or `CONTRIBUTING.md`) or put in a more general developer guide like [SciPy](https://docs.scipy.org/doc/scipy/reference/dev/index.html#making-a-scipy-release) does?

**Comment by ftalbrecht** [comment]
```
- [ ] Update authors and contributors
```

**Comment by gramian** [comment]
What about Researchgate and NA-digest?

**Comment by renefritze** [comment]
> ```
> - [ ] Update authors and contributors
> ```

We currently ask this in PRs, so that should be OK.

 > BTW, why not use a numbered list?

Actually didn't use a list of any kind because I thought this didn't render as clickable.

**Comment by pmli** [comment]
> Actually didn't use a list of any kind because I thought this didn't render as clickable.

Right now it renders as a single line...

File: RELEASE_CHECKLIST.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+ 1. [ ] (Create release branch)
+ 1. [ ] With each 20XX.1 release, update Copyright notices accordingly
+ 1. [ ] Check wheel install on debian stretch produces an OK error message due to too old python
+ 1. [ ] Write Release Notes
+ 1. [ ] Update Readme
+ 1. [ ] Update Copyright notice if year roll over
```
Comment (sdrave) [inline RELEASE_CHECKLIST.md]: How is this different from 'update Copyright notices accordingly'?

File: RELEASE_CHECKLIST.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+
+ 1. [ ] (Create release branch)
+ 1. [ ] With each 20XX.1 release, update Copyright notices accordingly
+ 1. [ ] Check wheel install on debian stretch produces an OK error message due to too old python
+ 1. [ ] Write Release Notes
+ 1. [ ] Update Readme
```
Comment (sdrave) [inline RELEASE_CHECKLIST.md]: Update Homepage

File: RELEASE_CHECKLIST.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ 1. [ ] Update Copyright notice if year roll over
+ 1. [ ] Tag commit
+ 1. [ ] Push tag
+ 1. [ ] bump/create demo docker
+ 1. [ ] Wait for CI build to finish (this will upload sdist+wheels to pypi, conda builds to conda cloud)
+ 1. [ ] update morwiki https://morwiki.mpi-magdeburg.mpg.de/morwiki/index.php/PyMOR
+ 1. [ ] update researchgate https://www.researchgate.net/project/pyMOR-Sustainable-Software-for-Model-Order-Reduction
```
Comment (sdrave) [inline RELEASE_CHECKLIST.md]: I thought we would delete the researchgate project for the DFG project?

File: RELEASE_CHECKLIST.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+ 1. [ ] (Create release branch)
+ 1. [ ] With each 20XX.1 release, update Copyright notices accordingly
+ 1. [ ] Check wheel install on debian stretch produces an OK error message due to too old python
+ 1. [ ] Write Release Notes
+ 1. [ ] Update Readme
+ 1. [ ] Update Copyright notice if year roll over
+ 1. [ ] Tag commit
+ 1. [ ] Push tag
+ 1. [ ] bump/create demo docker
+ 1. [ ] Wait for CI build to finish (this will upload sdist+wheels to pypi, conda builds to conda cloud)
```
Comment (sdrave) [inline RELEASE_CHECKLIST.md]: Zenodo release

Is the github release automatic?

Comment (sdrave) [inline RELEASE_CHECKLIST.md]: Send release announcement to pymor-dev

File: RELEASE_CHECKLIST.md (hunk 3 above)
Comment (renefritze) [reply RELEASE_CHECKLIST.md]: No, we deleted the non-DFG one.

File: RELEASE_CHECKLIST.md (hunk 4 above)
Comment (renefritze) [reply RELEASE_CHECKLIST.md]: It's the other way round.

File: RELEASE_CHECKLIST.md (hunk 3 above)
Comment (sdrave) [reply RELEASE_CHECKLIST.md]: @pymor/pymor-devs, I was quite sure we agreed to delete the DFG-one?! What would be the argument for having a DFG-project related thing on researchgate but not for pyMOR itself?

Comment (pmli) [reply RELEASE_CHECKLIST.md]: I don't remember what was said in the dev meeting, but I would also be for deleting the DFG project and having a general one.

Comment (renefritze) [reply RELEASE_CHECKLIST.md]: We now have the the general one here: https://www.researchgate.net/project/pyMOR-Model-Order-Reduction-with-Python
Sorry for my mix-up.

**Comment by pmli** [comment]
It is still not clear to me why should this document appear in the home directory of the project when it only concerns release managers... I would rather have it in `docs/`.

**Pull request merged**